### PR TITLE
Move entity filter "edit" button to a section header icon

### PR DESCRIPTION
**Pull request by abey79**
### What

* Builds on #6657
* Part of #6655 

This PR is mainly a refactor that abstract the buttons used by `PropertyContent` and `SectionCollapsingHeader`. (This refactor is partial and should also be applied to `LabelContent` -> #6191).

This enables action button in section header, used to remove the "edit" button of the entity filter:
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6662?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6662?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6662)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_ui/src/section_collapsing_header.rs [hunk 1]
```diff
@@ -1,75 +1,11 @@
-use crate::{list_item, DesignTokens, Icon, UiExt as _};
-
-/// Icon button to be used in the header of a panel.
-pub struct HeaderMenuButton<'a> {
```
Comment (Wumpf) [inline crates/re_ui/src/section_collapsing_header.rs:4]: didn't I review this being moved just a few hours ago :smile:

File: crates/re_ui/src/list_item/property_content.rs [hunk 2]
```diff
@@ -37,7 +20,8 @@ pub struct PropertyContent<'a> {
     show_only_when_collapsed: bool,
     value_fn: Option<Box<PropertyValueFn<'a>>>,
     //TODO(ab): in the future, that should be a `Vec`, with some auto expanding mini-toolbar
-    property_buttons: Option<PropertyButton<'a>>,
+    button: Option<Box<dyn super::ItemButton + 'a>>,
+    //property_buttons: Option<PropertyButton<'a>>,
```
Comment (Wumpf) [inline crates/re_ui/src/list_item/property_content.rs:24]: leftover?

**Review by Wumpf — APPROVED** [review]
so pretty! both the ui and the code improvement of having an `ItemButton` concept

File: crates/re_ui/src/list_item/property_content.rs (hunk 2 above)
Comment (abey79) [reply crates/re_ui/src/list_item/property_content.rs:24]: yup!

File: crates/re_ui/src/section_collapsing_header.rs (hunk 1 above)
Comment (abey79) [reply crates/re_ui/src/section_collapsing_header.rs:4]: That pesky thing had a short—but intense—life :)

**Pull request merged**